From 86ffe5425b4261a04fea6cfadc2c188776e05466 Mon Sep 17 00:00:00 2001
From: Arseniy Velikanov <adomerlee@gmail.com>
Date: Wed, 16 Aug 2023 08:06:07 +0400
Subject: [PATCH 39/91] iommu/mediatek: Add support for MT6765 M4U

Signed-off-by: Arseniy Velikanov <adomerlee@gmail.com>
---
 drivers/iommu/mtk_iommu.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/iommu/mtk_iommu.c b/drivers/iommu/mtk_iommu.c
index 0e0285348..d0308c8b3 100644
--- a/drivers/iommu/mtk_iommu.c
+++ b/drivers/iommu/mtk_iommu.c
@@ -165,6 +165,7 @@
 
 enum mtk_iommu_plat {
 	M4U_MT2712,
+	M4U_MT6765,
 	M4U_MT6779,
 	M4U_MT6795,
 	M4U_MT8167,
@@ -1509,6 +1510,17 @@ static const struct dev_pm_ops mtk_iommu_pm_ops = {
 				     pm_runtime_force_resume)
 };
 
+static const struct mtk_iommu_plat_data mt6765_data = {
+	.m4u_plat       = M4U_MT6765,
+	.flags          = RESET_AXI | MTK_IOMMU_TYPE_MM,
+	.inv_sel_reg    = REG_MMU_INV_SEL_GEN1,
+	.banks_num      = 1,
+	.banks_enable   = {true},
+	.iova_region    = single_domain,
+	.iova_region_nr = ARRAY_SIZE(single_domain),
+	.larbid_remap   = {{0}, {1}, {2}, {3}},
+};
+
 static const struct mtk_iommu_plat_data mt2712_data = {
 	.m4u_plat     = M4U_MT2712,
 	.flags        = HAS_4GB_MODE | HAS_BCLK | HAS_VLD_PA_RNG | SHARE_PGTABLE |
@@ -1786,6 +1798,7 @@ static const struct mtk_iommu_plat_data mt8365_data = {
 
 static const struct of_device_id mtk_iommu_of_ids[] = {
 	{ .compatible = "mediatek,mt2712-m4u", .data = &mt2712_data},
+	{ .compatible = "mediatek,mt6765-m4u", .data = &mt6765_data},
 	{ .compatible = "mediatek,mt6779-m4u", .data = &mt6779_data},
 	{ .compatible = "mediatek,mt6795-m4u", .data = &mt6795_data},
 	{ .compatible = "mediatek,mt6893-iommu-mm", .data = &mt6893_data},
