From e60cf0ba96194194acab655fa78644c4dfa3264b Mon Sep 17 00:00:00 2001
From: Arseniy Velikanov <adomerlee@gmail.com>
Date: Sun, 20 Aug 2023 23:13:28 +0400
Subject: [PATCH 66/91] clk: mediatek: mt6765-cam: Add reset support

Signed-off-by: Arseniy Velikanov <adomerlee@gmail.com>
---
 drivers/clk/mediatek/clk-mt6765-cam.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/clk/mediatek/clk-mt6765-cam.c b/drivers/clk/mediatek/clk-mt6765-cam.c
index 2a7f30dc8..4edd3e378 100644
--- a/drivers/clk/mediatek/clk-mt6765-cam.c
+++ b/drivers/clk/mediatek/clk-mt6765-cam.c
@@ -33,9 +33,18 @@ static const struct mtk_gate cam_clks[] = {
 	GATE_CAM(CLK_CAM_CCU, "cam_ccu", "mm_ck", 12),
 };
 
+static u16 cam_rst_ofs[] = { 0xc };
+
+static const struct mtk_clk_rst_desc cam_resets = {
+	.version = MTK_RST_SIMPLE,
+	.rst_bank_ofs = cam_rst_ofs,
+	.rst_bank_nr = ARRAY_SIZE(cam_rst_ofs)
+};
+
 static const struct mtk_clk_desc cam_desc = {
 	.clks = cam_clks,
 	.num_clks = ARRAY_SIZE(cam_clks),
+	.rst_desc = &cam_resets,
 };
 
 static const struct of_device_id of_match_clk_mt6765_cam[] = {
