From 9d77afa8513929d39e27982fea3dfada2e948d59 Mon Sep 17 00:00:00 2001
From: Arseniy Velikanov <adomerlee@gmail.com>
Date: Sun, 20 Aug 2023 23:25:50 +0400
Subject: [PATCH 69/91] clk: mediatek: mt6765-vcodec: Add reset support

Signed-off-by: Arseniy Velikanov <adomerlee@gmail.com>
---
 drivers/clk/mediatek/clk-mt6765-vcodec.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/clk/mediatek/clk-mt6765-vcodec.c b/drivers/clk/mediatek/clk-mt6765-vcodec.c
index f309d1090..180e0f564 100644
--- a/drivers/clk/mediatek/clk-mt6765-vcodec.c
+++ b/drivers/clk/mediatek/clk-mt6765-vcodec.c
@@ -28,9 +28,18 @@ static const struct mtk_gate venc_clks[] = {
 	GATE_VENC(CLK_VENC_SET3_VDEC, "venc_set3_vdec", "mm_ck", 12),
 };
 
+static u16 venc_rst_ofs[] = { 0xc };
+
+static const struct mtk_clk_rst_desc venc_resets = {
+	.version = MTK_RST_SIMPLE,
+	.rst_bank_ofs = venc_rst_ofs,
+	.rst_bank_nr = ARRAY_SIZE(venc_rst_ofs)
+};
+
 static const struct mtk_clk_desc venc_desc = {
 	.clks = venc_clks,
 	.num_clks = ARRAY_SIZE(venc_clks),
+	.rst_desc = &venc_resets,
 };
 
 static const struct of_device_id of_match_clk_mt6765_vcodec[] = {
